### PR TITLE
Adding support for configuring ...

### PR DESCRIPTION
**Pull request by iuliana**
... ssh connection details using connection tags.

### CODE REVIEW

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java [hunk 1]
```diff
@@ -1315,4 +1314,79 @@ public void testAddPolicyWithDslFromDeployableBlueprint() throws Exception {
         EntityAsserts.assertAttributeEquals(entity, Sensors.newSensor(Object.class, "result"), "yes");
     }
 
+    @Test
```
Comment (ahgittin) [inline camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java:1317]: any test (such as this) that actually performs SSH should be marked as `@Test(groups="Integration")` as unit tests shouldn't assume the ability to `ssh localhost`

alternatively it a MockSshLocation can be used (add something like `mock-location` to catalog and pass it)

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java [hunk 2]
```diff
@@ -1315,4 +1314,79 @@ public void testAddPolicyWithDslFromDeployableBlueprint() throws Exception {
...
+                "steps:",
+                "- type: ssh\n" +
+                "  command: |\n" +
+                "    echo \"init-done\" >> wf.log"), "test");
+        Object result  = x1.getTask(false).get().get();
+        Asserts.assertEquals(((Map)result).get("exit_code"), 0);
+    }
+
+    @Test
+    public void testSshStepOnLocalhostDefinition() throws Exception {
```
Comment (ahgittin) [inline camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java:1337]: remove `lock: x`

make integration test - or support mocks for connection definition.  that seems harder, but quite useful.  i think another mechanism we use is to specify the `RecordingSshTool`, so if there is a way to supply a tool class that could work.  (the recording tool does nothing and returns 0 for all ssh commands, but just records the commands.)

**Comment by ahgittin** [comment]
looks good.

some of the tests need tidying.

the test failure reported is

```
testWorkflowLogging(org.apache.brooklyn.core.workflow.WorkflowBasicTest)  Time elapsed: 0.064 sec  <<< FAILURE!
java.lang.AssertionError: null: lists don't have the same size expected [8] but found [11]
	at org.apache.brooklyn.core.workflow.WorkflowBasicTest.testWorkflowLogging(WorkflowBasicTest.java:381)
```

which is almost certainly unrelated.  i think i've seen it before.  not sure why it happens occasionally (nice to fix that but not relevant here.)

merge after minor test fixes

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java (hunk 1 above)
Comment (iuliana) [reply camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java:1317]: @ahgittin I had this in the back of my mind to do before submitting the PR and I forgot. I annotated the test methods.

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java [hunk 3]
```diff
@@ -1315,4 +1314,79 @@ public void testAddPolicyWithDslFromDeployableBlueprint() throws Exception {
...
+    public void testSshStepOnLocalhostLocation() throws Exception {
+        String yaml =
+                "location: localhost\n" +
+                "services:\n" +
+                "  - type: " + WorkflowSoftwareProcess.class.getName() +"\n" +
+                "    name: sample-server\n" ;
+        Entity app = createStartWaitAndLogApplication(yaml);
+
+        WorkflowExecutionContext x1 = WorkflowBasicTest.runWorkflow(app.getChildren().iterator().next(), Strings.lines(
+                "lock: x",
```
[the remark this answers is not in the document]
Comment (iuliana) [reply camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java:1327]: Done.

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java (hunk 2 above)
Comment (iuliana) [reply camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java:1337]: Done.

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java [hunk 4]
```diff
@@ -1315,4 +1314,79 @@ public void testAddPolicyWithDslFromDeployableBlueprint() throws Exception {
...
+                "steps:",
+                "- type: ssh\n" +
+                        "  command: |\n" +
+                        "    echo \"init-done\" >> wf.log"), "test");
+        Object result  = x1.getTask(false).get().get();
+        Asserts.assertEquals(((Map)result).get("exit_code"), 0);
+    }
+
+    @Test
+    public void testSshStepOnLocalhostDefinitionWithExternalConfig() throws Exception {
```
[the remark this answers is not in the document]
Comment (iuliana) [reply camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java:1363]: Done.

**Pull request merged**